### PR TITLE
#1273 no js issue

### LINKED ISSUE #1179: Sidemenu incomplete after Shibboleth login
**Describe the bug**
When a user with administrative permission log in via shibboleth, the sidebar is populated with an incomplete items set. After a page refresh the sidebar is then populated correctly.

**To Reproduce**
1. Perform the login via shibboleth with a user with administrative permission (ex sheldon/bazinga).
sheldon1

2. Check the sidemenu content
sheldon2
3. Perform a page refresh
sheldon3

**Expected behavior**
The sidemenu should be populated correctly after the successful login redirect.

### LINKED ISSUE #1273: UI does not load when JavaScript is disabled in browser
**Describe the bug**
If you turn off Javascript in your browser, the UI becomes unresponsive.  No errors appear (at least not in the DevTools in the browser), but no page will load successfully.

**UPDATE: I've determined this is caused by #1244** If I revert that PR, everything works. 

**To Reproduce**
1. Turn off JS in your browser tab (e.g in Chrome, go into DevTools, select "Settings" (upper right), check "Disable Javascript")
2. Visit the demo site (any page), e.g. https://demo7.dspace.org/  in that tab.  This is also reproducible when running the UI on localhost:4000 (via `yarn start`)

**Expected behavior**
For SEO purposes, the site must work (at least public pages) with Javascript disabled.

### PR DESCRIPTION
**Pull request by atarix83**
## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes #1273 
* Fixes #1179

## Description
This PR resolves both issues mentioned previously. I've reverted the https://github.com/DSpace/dspace-angular/pull/1244 in order to fix the #1273 and then I've used a new approach to solve the #1179.
Since the #1179 issue was due to cache problem with request to `authorizations` endpoint. I've added a method to set stale all the requests to the  `authorizations` endpoint when the store is rehydrated from SSR to CSR

**Include guidance for how to test or review your PR.** 
Test both issues are sorted out.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

File: src/app/core/cache/object-cache.effects.ts [hunk 1]
```diff
@@ -18,10 +19,11 @@ export class ObjectCacheEffects {
    */
   @Effect() fixTimestampsOnRehydrate = this.actions$
     .pipe(ofType(StoreActionTypes.REHYDRATE),
-      map(() => new ResetObjectCacheTimestampsAction(new Date().getTime()))
+      map(() => new ResetObjectCacheTimestampsAction(new Date().getTime())),
+      tap(() => this.authorizationsService.invalidateAuthorizationsRequestCache())
```
Comment (artlowel) [inline src/app/core/cache/object-cache.effects.ts:23]: I wouldn't add this to the bottom of another, unrelated effect. I'd add a separate effect that listens to the same action, with `dispatch: false` especially for this. 

That way the effect can have a name that reflects what it does, it makes the code easier to understand, and to describe in typedocs.

**Review by artlowel — CHANGES_REQUESTED** [review]
Thanks @atarix83!

I can verify that it fixes #1273, and still fixes #1179

I just have one code comment: For clarity I wouldn't append this to the existing effect but rather make a separate one.

File: src/app/core/cache/object-cache.effects.ts (hunk 1 above)
Comment (artlowel) [reply src/app/core/cache/object-cache.effects.ts:23]: That way it can also be moved to auth.effects.ts which is likely a better fit

**Review by tdonohue — APPROVED** [review]
👍  @atarix83 : Thanks for the fix here! I've tested this thoroughly and I can verify it fixes both #1273 and #1179.

So, as soon as @artlowel 's feedback is addressed, I think this can be merged.

File: src/app/core/cache/object-cache.effects.ts (hunk 1 above)
Comment (atarix83) [reply src/app/core/cache/object-cache.effects.ts]: done

**Comment by tdonohue** [comment]
👍 Thanks @atarix83 !  I've retested today and this is still working after the refactor. It also looks (to me) that you've addressed Art's feedback.

I'll let @artlowel give this one final review tomorrow (as he's out today) and then we'll get this merged.

**Review by artlowel — APPROVED** [review]
Thanks @atarix83!

**Pull request merged**